.github/workflows/snapshot-docker-version-tag.yml

```diff
@@ -0,0 +1,40 @@
+# description: Generate a Docker version tag for snapshot builds based on the branch name.
+# type: CI
+# owner: @camunda/monorepo-devops-team
+name: Generate Docker Version Tag (SNAPSHOT)
+
+permissions:
+  contents: read
+
+on:
+  workflow_call:
+    outputs:
+      version_tag:
+        description: "Generated Docker version tag"
+        value: ${{  jobs.get-snapshot-docker-version-tag.outputs.tag }}
+
+env:
+  GHA_BEST_PRACTICES_LINTER: enabled
+
+jobs:
+  get-snapshot-docker-version-tag:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    permissions: {}
+    if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/heads/stable/')
```

Comment:
I think having an additional condition `github.repository == 'camunda/camunda'` is useful here, since the monorepo is public and this code could also be running in forked repositories - where it is not needed since those forked repos don't deploy any Docker images.